hystrix-contrib/hystrix-javanica/hystrix-javanica-core/src/main/java/com/netflix/hystrix/contrib/javanica/aop/AbstractCommandMetaHolderBuilder.java

```diff
@@ -0,0 +1,220 @@
+
+		builder.defaultCommandKey(batchCommandMethod.getName());
+		builder.hystrixCommand(hystrixCommand);
+		builder.executionType(ExecutionType.getExecutionType(batchReturnType));
+		builder.observable(observable);
+		FallbackMethod fallbackMethod = MethodProvider.getInstance().getFallbackMethod(proxyClass, batchCommandMethod);
+		if (fallbackMethod.isPresent()) {
+			fallbackMethod.validateReturnType(batchCommandMethod);
+			builder.fallbackMethod(fallbackMethod.getMethod()).fallbackExecutionType(ExecutionType.getExecutionType(fallbackMethod.getMethod().getReturnType()));
+		}
+		customizeCollapserBuilder(hystrixCollapser, batchCommandMethod);
+
+	}
+
+	protected void customizeCollapserBuilder(HystrixCollapser hystrixCollapser, Method batchCommandMethod) {
+
+	}
+
+	protected void customizeCommandBuilder(HystrixCommand hystrixCommand) {
+
+	}
+	
+	protected void  setDefaultProperties() {
+		Optional<DefaultProperties> defaultPropertiesOpt = AopUtils.getAnnotation(proxyClass, DefaultProperties.class);
+		builder.defaultGroupKey(proxyClass.getSimpleName());
+		if (defaultPropertiesOpt.isPresent()) {
+			DefaultProperties defaultProperties = defaultPropertiesOpt.get();
+			builder.defaultProperties(defaultProperties);
+			if (StringUtils.isNotBlank(defaultProperties.groupKey())) {
+				builder.defaultGroupKey(defaultProperties.groupKey());
+			}
+			if (StringUtils.isNotBlank(defaultProperties.threadPoolKey())) {
+				builder.defaultThreadPoolKey(defaultProperties.threadPoolKey());
+			}
+		}
+		
+	}
+
+	protected  void setFallbackMethod() {
+		FallbackMethod fallbackMethod = MethodProvider.getInstance().getFallbackMethod(proxyClass, method);
+		if (fallbackMethod.isPresent()) {
+			fallbackMethod.validateReturnType(method);
+			builder.fallbackMethod(fallbackMethod.getMethod()).fallbackExecutionType(ExecutionType.getExecutionType(fallbackMethod.getMethod().getReturnType()));
+		}
+		
+	}
+	protected Method getBatchCommandMethod(HystrixCollapser collapser){
+		return getDeclaredMethod(proxyClass, collapser.batchMethod(), List.class);
+	}
+	protected Class<?> getFirstGenericParameter(Type type) {
+		return getFirstGenericParameter(type, 1);
+	}
+
+
+	protected Class<?> getFirstGenericParameter(final Type type, final int nestedDepth) {
+		int cDepth = 0;
+		Type tType = type;
+
+		for (int cDept = 0; cDept < nestedDepth; cDept++) {
+			if (!(tType instanceof ParameterizedType))
+				throw new IllegalStateException(String.format("Sub type at nesting level %d of %s is expected to be generic", cDepth, type));
+			tType = ((ParameterizedType) tType).getActualTypeArguments()[cDept];
+		}
+
+		if (tType instanceof ParameterizedType)
+			return (Class<?>) ((ParameterizedType) tType).getRawType();
+		else if (tType instanceof Class)
+			return (Class<?>) tType;
+
+		throw new UnsupportedOperationException("Unsupported type " + tType);
+	}
+	public enum HystrixPointcutType {
+		COMMAND, COLLAPSER;
+
```

Comment:
I use eclipse. It looks like there is no straight forward way to import intellij formatter to eclipse. Let me find some work around.